### PR TITLE
UpdateOnPlayerAdded for lists and optimized commands

### PR DESCRIPTION
**Pull request by LolloDev5123**
Better serverdetails and stuff yessings???
serverdetails now have serverinfo and serverdetails as aliases instead of :details which was a bit confusing admit it
please make sure there are no vulnerabilities on the event that checks player added in the list module, im not big expert, does it trigger on nil players when they are "added" too?

RobloxScreenShot20210317_035622478

Also as a suggestion we need an autosize button for lists that automatically sizes them to the biggest possible size  that doesn't make the list small as my pp

### CODE REVIEW

File: MainModule/Server/Core/Logs.lua [hunk 1]
```diff
@@ -39,21 +38,27 @@ return function(Vargs)
 		Chats = {};
 		Joins = {};
 		Script = {};
+		Replications = {};
+		NetworkOwners = {};
```
Comment (ccuser44) [inline MainModule/Server/Core/Logs.lua:42]: Why are replications and entwork owners added. They were remvoed before and I suggest you remove them.

File: MainModule/Server/Core/Logs.lua [hunk 2]
```diff
@@ -39,21 +38,27 @@ return function(Vargs)
...
 		TempUpdaters = {};
-		
 		TabToType = function(tab)
 			local indToName = {
 				Chats = "Chat";
 				Joins = "Join";
 				Script = "Script";
+				Replications = "Replication";
+				PrivateMessages = "PrivateMessages";
+				NetworkOwners = "NetworkOwner";
```
Comment (ccuser44) [inline MainModule/Server/Core/Logs.lua:56]: Same. Do not add replications and network owner

File: MainModule/Server/Core/Logs.lua [hunk 3]
```diff
@@ -207,6 +213,22 @@ return function(Vargs)
...
+			end;
+			
+			NetworkOwners = function()
+				local tab = {}
+				for i,v in pairs(Logs.NetworkOwners) do
+					table.insert(tab,{Text = tostring(v.Player).." made "..tostring(v.Part),Desc = v.Path})
+				end
+				return tab
+			end;
+
```
Comment (ccuser44) [inline MainModule/Server/Core/Logs.lua:231]: Do not add replications and network owners. They were removed in the removal of FD spesific code.

File: MainModule/Server/Core/Logs.lua [hunk 4]
```diff
@@ -199,6 +204,7 @@ return function(Vargs)
 				return temptable
 			end;
 			
+
```
Comment (ccuser44) [inline MainModule/Server/Core/Logs.lua:207]: Extra line?

File: MainModule/Server/Core/Logs.lua [hunk 5]
```diff
@@ -39,21 +38,27 @@ return function(Vargs)
 		Chats = {};
 		Joins = {};
 		Script = {};
+		Replications = {};
+		NetworkOwners = {};
 		RemoteFires = {};
 		Commands = {};
 		Exploit = {};
 		Errors = {};
+		ServerDetails = {};
```
Comment (ccuser44) [inline MainModule/Server/Core/Logs.lua:47]: Why is ServerDetails added to the logs?

File: MainModule/Server/Core/Logs.lua [hunk 6]
```diff
@@ -39,21 +38,27 @@ return function(Vargs)
...
 				Joins = "Join";
 				Script = "Script";
+				Replications = "Replication";
+				PrivateMessages = "PrivateMessages";
+				NetworkOwners = "NetworkOwner";
 				RemoteFires = "RemoteFire";
 				Commands = "Command";
 				Exploit = "Exploit";
 				Errors = "Error";
+				ServerDetails = "ServerDetails"
```
Comment (ccuser44) [inline MainModule/Server/Core/Logs.lua:61]: Server details in the logs?

**Review by ccuser44 — CHANGES_REQUESTED** [review]
Nice pull request. 😄 
However there are some issues and weird stuff like network owner logs and replication logs being readded.

File: MainModule/Server/Core/Logs.lua (hunk 2 above)
Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:56]: Also wut is 
				PrivateMessages = "PrivateMessages";

**Comment by SimTheProtogen** [comment]
~~ok lol~~

**Comment by EnderUwU** [comment]
[new pogchamp adonis feature i made lol its poggers ](https://youtu.be/bxqLsrlakK8)

File: MainModule/Server/Commands/Moderators.lua [hunk 7]
```diff
@@ -1957,12 +1969,14 @@ return function(Vargs, env)
 			AdminLevel = "Moderators";
 			Function = function(plr,args)
 				local question = args[3]
-				if not question then error("You forgot to supply a question!") end
+				assert(question, "Question not supplied")
 				local answers = args[2]
+				assert(answers, "Answers not supplied")
```
[the remark this answers is not in the document]
Comment (LolloDev5123) [reply MainModule/Server/Commands/Moderators.lua:1974]: Oh yeah, I guess I have a kinda old fork

File: MainModule/Server/Core/Logs.lua (hunk 6 above)
Comment (LolloDev5123) [reply MainModule/Server/Core/Logs.lua:61]: I think you are sending too many comments about the same thing? Server Details are now as a log to better access the table Logs.ServerDetails

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:56]: Yeah I think I have.

File: MainModule/Server/Core/Logs.lua [hunk 8]
```diff
@@ -282,62 +283,67 @@ return function(Vargs)
...
+				else
+					table.insert(tab,{Text = "Loadstring: [OFF]"})
+
+				end
+
+				if service.Workspace.StreamingEnabled == true then
+					table.insert(tab,{Text = "Streaming: [ON]"})
+				else
+					table.insert(tab,{Text = "Streaming: [OFF]"})
 				end
```
Comment (ccuser44) [inline MainModule/Server/Core/Logs.lua:336]:
```
				table.insert(tab,{Text = "HTTPService: ["..HTTP.CheckHttp() and "ON" or OFF.."]"})
				table.insert(tab,{Text = "Third Party Sales: ["..game.Workspace.AllowThirdPartySales and "ON" or OFF.."]"})

				table.insert(tab,{Text = "Loadstring: ["..pcall(function() loadstring("local hi = 'test'") end)  and "ON" or "OFF".."]"})

				table.insert(tab,{Text = "Streaming: ["..service.Workspace.StreamingEnabled and "ON" or OFF.."]"})
```

**Review by ccuser44 — CHANGES_REQUESTED** [review]
Some things to optimise further ;)

**Comment by ccuser44** [comment]
You can mark my previous review as solved.

File: MainModule/Server/Core/Logs.lua (hunk 8 above)
Comment (LolloDev5123) [reply MainModule/Server/Core/Logs.lua:336]: Im not a LUA Expert, did you test this?

File: MainModule/Server/Commands/Moderators.lua [hunk 9]
```diff
@@ -5653,20 +5664,15 @@ return function(Vargs, env)
...
-						--table.insert(temp,{Text=v.message,Desc='Error: '..v.message,Color=Color3.new(1,0,0)})
+					if v.messageType == Enum.MessageType.MessageOutput then
+						toTab(v.message, "Output - ")
+					elseif v.messageType == Enum.MessageType.MessageWarning then
+						toTab(v.message, "Warning - ", Color3.new(0.866667, 0.733333, 0.0509804))
+					elseif v.messageType == Enum.MessageType.MessageInfo then
+						toTab(v.message, "Info - ", Color3.new(0.054902, 0.305882, 1))
+					elseif v.messageType == Enum.MessageType.MessageError then
+						toTab(v.message, "Error - ", Color3.new(1, 0.196078, 0.054902))
 					end
```
[the remark this answers is not in the document]
Comment (LolloDev5123) [reply MainModule/Server/Commands/Moderators.lua:5676]: WOW, cool stuff.

File: MainModule/Server/Core/Logs.lua (hunk 8 above)
Comment (LolloDev5123) [reply MainModule/Server/Core/Logs.lua:336]: I tested it, I found that OFF wasnt written as a string, I fixed it, and it's still giving me, Attempt to concactenate string with nil

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: Oh yes I just forgot to add quotes.
It should be like "OFF"

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: Also at the other place I forgot to add quotes (The first comment of this change request)

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: I added them to the comments.

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: Wait wait I also forgot to add it in brackets. Ill fix that.

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: Fixed it. Not it should work.

Comment (LolloDev5123) [reply MainModule/Server/Core/Logs.lua:336]: Alright, I'll test this out!

Comment (LolloDev5123) [reply MainModule/Server/Core/Logs.lua:336]: It's still giving me the Attempt to concactenate string with nill

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: Can you show me the code that puts out the error?

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: Like the exact part

Comment (LolloDev5123) [reply MainModule/Server/Core/Logs.lua:336]: The first line that displays the HTTPService, so it could be all of the lines

Comment (ccuser44) [reply MainModule/Server/Core/Logs.lua:336]: Hmm... I have to test it when I can

**Comment by LolloDev5123** [comment]
i am too lazy to fix conflicts and i am going to make small prs

**Pull request closed without merging**